### PR TITLE
feat(mergerequests): publish 15 fields and declare 34 the server never asks for

### PR DESCRIPTION
**Pull request by jmrplens**
## Description

The sent dimension named 51 fields across the merge request family, and the interesting half of this layer is that only fifteen of them are real.

Thirty-four are one artefact with one cause, and it is worth stating because it is the class that produced the confirmed phantom of [issue 580](https://github.com/jmrplens/gitlab-mcp-server/issues/580). The audit pairs our merge request output with client-go's merge request structs and then unions every endpoint those structs' methods reach. Two of those methods answer with something else entirely: `ChangeApprovalConfiguration` on the approvals service returns an approval state, and the deprecated `GetMergeRequestChanges` returns a changes object. This server calls neither, and publishing what they send would have invented fifteen approval counters on every row of a merge request list. They are answered by a new declaration category rather than by a field.

The two rendered-HTML fields turned out sharper than the review expected. `render_html` is not a presenter option nobody can set: it is a declared parameter on the single merge request GET, and `mr.get` already accepts it, already sends it to GitLab, and then discards the answer. Publishing `title_html` and `description_html` is what makes a parameter we were already sending mean something. The issue routes do not declare that parameter, so the same two keys are declared there instead of published.

One field was a phantom of our own. `work_in_progress` came from the SDK struct, which models it on one type only, so every list published the key and filled it from nothing. It now comes from the captured response, and `search`, which aliases the same output type, is wired with it for the same reason.

This is the top of the stack, so the freshness artifacts are regenerated here.

## Related Issue

No single issue: this is the merge request tranche of the continuing 1:1 field review, whose upstream half is tracked in <https://gitlab.com/gitlab-org/api/client-go/-/issues/2300>.

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [x] Enhancement (improvement to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring (no functional change)
- [ ] Documentation update
- [ ] CI / build / tooling

## Changes Made

- Publishes fifteen merge request fields GitLab sends and client-go does not model, read from the captured response under [ADR-0021](docs/development/adr/adr-0021-captured-response-for-fields-the-sdk-does-not-model.md).
- Adds the `sdk-route-no-handler-calls` declaration category and answers 34 findings with it: an entity reached only through an SDK method no handler in this repository calls.
- Fills `work_in_progress` from the capture instead of the SDK struct, which modelled it on one type and left the key empty on every list.
- Wires the same shape through `search`, which aliases the merge request output type, so the alias cannot publish a key nothing fills.
- Declares `title_html` and `description_html` on the issue routes, which do not declare the parameter that produces them, and publishes them on the merge request route, which does.
- Fixes one Sonar S8193 smell at the tip of the stack rather than relaunching the layer below it.
- Regenerates the freshness artifacts, which a stack refreshes once at its top.

## How to Test

1. `go run ./cmd/audit_1to1/ -scope=paths 2>&1` and read `summary`: `typed_unsurfaced_fields` falls 1367 to 1352, `typed_unsurfaced_declared` rises 1073 to 1109, the undeclared backlog falls 294 to 243, and `stale_declarations` is 0.
2. In the same report, `mergerequests`, `deploymentmergerequests`, `mrapprovals`, `mrchanges` and `search` all read 0 undeclared.
3. `go test ./internal/tools/mergerequests/ ./internal/tools/search/ ./internal/tools/geo/ -count=1`.
4. `make check-stats check-llms check-footprint check-site-stats check-testing-docs`.

## Breaking Changes / Migration Notes

N/A. Every field is additive, and the one field whose source changed (`work_in_progress`) was previously published empty on every list.

### Code Quality

- [x] Code compiles: `go build ./...`
- [x] Consolidated Go analysis passes on changed packages
- [x] Code is formatted: `make analyze-fix`
- [x] Follows idiomatic Go patterns and the conventions documented in `CLAUDE.md` / `.github/instructions/`

### Testing

- [x] All existing tests pass
- [x] New tests added for new functionality (table-driven, with `httptest` mocks)
- [x] Coverage on modified packages is at project policy
- [x] Edge cases and error scenarios covered
- [ ] If applicable, E2E tests updated/added under `test/e2e/suite/`

### Documentation

- [x] Doc comments added for exported types/functions (godoc)
- [x] `docs/` updated if public API or behavior changed
- [x] `README.md` / Starlight site updated if user-facing behavior changed
- [ ] If introducing/removing a tool: `docs/tools/{domain}.md` and tool counts updated
- [x] Commit messages follow [Conventional Commits](https://www.conventionalcommits.org/)

### Security

- [x] No secrets, tokens, or credentials in code, tests, fixtures, or logs
- [x] Input validation for user-provided parameters
- [x] Error messages do not leak sensitive information
- [x] No new dependencies with known vulnerabilities

### CODE REVIEW

File: internal/tools/deploymentmergerequests/deployment_merge_requests_test.go [hunk 1]
```diff
@@ -906,3 +907,176 @@ func TestList_ApproverFilterInvalid_ReturnsError(t *testing.T) {
...
+}
+
+// TestList_EmptyApproverFilters_ReachNoQueryParameter verifies an unset
+// approver filter is left off the request rather than sent empty. The three
+// length guards in buildListOptions are what decides that, and a boundary
+// mutation of any of them turns "the caller asked for this" into "always".
+func TestList_EmptyApproverFilters_ReachNoQueryParameter(t *testing.T) {
+	client := testutil.NewTestClient(t, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		q := r.URL.Query()
+		if q.Has("approver_ids") || q.Has("approved_by_ids") || q.Has("approved_by_usernames") {
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply internal/tools/deploymentmergerequests/deployment_merge_requests_test.go:1015]: Half right, and the suggested diff would cost coverage rather than add it.

`ApproverIDsValue.EncodeValues` in client-go has two wire forms, chosen by the value the caller set. A `UserIDValue`, which is what `Any` and `None` become, is written with `v.Set(key, ...)` and so reaches the query as the bare `approver_ids`. A `[]int64` is written with `v.Add(key+"[]", ...)` and reaches it as `approver_ids[]`. Replacing the bare key with the bracketed one would stop this assertion covering the literal path entirely.

Both forms are already pinned exhaustively one level down, in `TestApproverIDsFilter_Encoding` in `internal/toolutil/approver_ids_test.go`: bracketed for numeric IDs, bare for `Any` and `None`, and nothing at all for an empty filter. That test owns the encoder's wire form; this one is a second guard at the handler.

The useful half of the finding stands, so I am taking it: the assertion will check all four spellings, both the bare and the bracketed key for each of the two ID filters, which costs nothing and makes the guard say what its comment already claims.

It lands in a follow-up rather than in this pull request. This is the top layer of a four-layer stack whose checks are nearly through, the change is test-only, and re-pushing here restarts every one of them for a guard that is not currently reachable in either spelling: an unset filter converts to a nil value that `omitempty` drops before the encoder is asked anything.
